### DISCUSSION TITLE
LFX Mentorship 2022 03-Sept-Nov Challenge - For OpenCV SDKs for Wasm

### DISCUSSION THREAD

**Discussion by alabulei1** [WasmEdge/WasmEdge#1754] [Show and tell]
For those who want to apply for LFX mentorship for #1747, this is a pretest for the application.

### Requirements

1. Research the WasmEdge QuickJS project, which builds a Rust wrapper around a C shared library. 

https://github.com/second-state/wasmedge-quickjs

2. Note that it uses the rust-bindgen tool to generate bindings from Rust to the shared library.

https://github.com/second-state/wasmedge-quickjs/tree/main/lib

3. **Your task:** Create a Rust wrapper for the OpenCV shared library. It must demonstrate how to call a function (any function you choose) from the OpenCV library. 

4. Post your repo's GitHub link as a comment under this discussion.

### Deadline
Please refer to [the timeline of LFX mentorship 2022 03-Sept-Nov](https://github.com/cncf/mentoring/tree/main/lfx-mentorship/2022/03-Sept-Nov#timeline).
The final date of valid submissions will be August 31, 5:00 PM PDT

**Comment by manulpatel** [comment]
Hello! I am trying to build the 1. WasmEdge QuickJS using my terminal. But when I enter `cargo build --target wasm32-wasi --release`, I am getting an errror in terminal as ``error: could not find `Cargo.toml` in `/home/manul/.cargo` or any parent directory``. Could someone please help?

**Reply by DawnMagnet** [reply]
You should learn some basic rust knowledge, and you must get into the wasmedge-quickjs folder to execute the cargo command

**Comment by omkar-mohanty** [comment]
Hello @alabulei1 
I finished the challenge

Link for Repo: [Link](https://github.com/omkar-mohanty/lfx-opencv-sdk)
I would love to improve the project and learn more. It was a bit challenging and I got to learn a lot.

Thanks
